### PR TITLE
Improve welcome message test

### LINKED ISSUE #252: Improve HydrateWelcomeMessageDictionary jUnit tests
## Task Context

Currently there is only one unit test for the [HydrateWelcomeMessageDictionary.java](https://github.com/jmakhack/hydrate-reminder/blob/master/src/main/java/com/hydratereminder/dictionary/HydrateWelcomeMessageDictionary.java) class and it is a very simplistic test that only checks if the return type of one of the methods is a String object.

```java
@Test
public void shouldGetRandomWelcomeMessage(){

    final String welcomeMessage = HydrateWelcomeMessageDictionary.getRandomWelcomeMessage();

    assertNotNull(welcomeMessage);
    assertFalse(welcomeMessage.isEmpty());
}
```

## Acceptance Criteria / Goals

The goal of this task is to improve the tests in [HydrateWelcomeMessageDictionaryTest.java](https://github.com/jmakhack/hydrate-reminder/blob/master/src/test/java/com/hydratereminder/dictionary/HydrateWelcomeMessageDictionaryTest.java).

One possible method of accomplishing this is to create a mock for the `SecureRandom.nextInt` method call. This will remove the randomness that occurs when trying to test the dictionary class.

See https://stackoverflow.com/questions/23375700/mocking-securerandomnextint for some idea on how this is done.

However this is only one possible solution and any other solutions to improving the tests would be much appreciated as well.

## Possible Alternatives

Any alternatives for improving the unit testing of [HydrateWelcomeMessageDictionary.java](https://github.com/jmakhack/hydrate-reminder/blob/master/src/main/java/com/hydratereminder/dictionary/HydrateWelcomeMessageDictionary.java) are definitely to be considered. For any discussions on how to improve the jUnit testing, join the discord server mentioned below. 

## Additional Notes

Please refer to [CONTRIBUTING.md](https://github.com/jmakhack/hydrate-reminder/blob/master/CONTRIBUTING.md) for info on how to setup, build, run, and test this project. Feel free to ping @jmakhack in the comments below with any questions if needed. Also, join the [Discord](https://discord.gg/RTgxfFW9mS) to meet and engage with other contributors!

[](https://discord.gg/RTgxfFW9mS)

### PR DESCRIPTION
**Pull request by Jacoberson**
## Associated Issue
Improve HydrateWelcomeMessageDictionaryTest by comparing a given String to an expected String from the HydrateWelcomeMessageDictionary
Closes #252

## Implemented Solution
Added PowerMock dependency and mocked SecureRandom.nextInt call and compared that String to a String from the HydrateWelcomeMessageDictionary

## Testing Details
Ran JUnit test locally

Operating System: Windows 11
RuneLite Version: 1.8.34.1

### CODE REVIEW

**Comment by jmakhack** [comment]
Hi @Jacoberson, some migration work to upgrade to JUnit 5 was just merged in. Can you resolve the conflicts and make sure your test still passes?

File: src/test/java/com/hydratereminder/dictionary/HydrateWelcomeMessageDictionaryTest.java [hunk 1]
```diff
@@ -1,18 +1,29 @@
...
+import org.mockito.Mockito;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
 
-import static org.junit.Assert.assertFalse;
-import static org.junit.Assert.assertNotNull;
+import java.security.SecureRandom;
 
+import static org.junit.Assert.*;
```
Comment (mslowiak) [inline src/test/java/com/hydratereminder/dictionary/HydrateWelcomeMessageDictionaryTest.java:12]: Please do not add the wildcard imports.

https://stackoverflow.com/questions/147454/why-is-using-a-wild-card-with-a-java-import-statement-bad

**Comment by Jacoberson** [comment]
@jmakhack I think I was able to figure out how to migrate the test to JUnit 5.
@mslowiak I removed the wildcard imports.

File: build.gradle [hunk 2]
```diff
@@ -32,4 +33,4 @@ sourceCompatibility = '1.8'
 
 tasks.withType(JavaCompile) {
 	options.encoding = 'UTF-8'
-}
```
Comment (jmakhack) [inline build.gradle:35]: Can you revert this change that removed the end of file newline?

**Review by jmakhack — APPROVED** [review]

**Comment by jmakhack** [comment]
@all-contributors please add @Jacoberson for tests
